File: src/main/scala/mffs/util/TransferMode.scala
package mffs.util

/**
 * The force field transfer mode.
 */
object TransferMode extends Enumeration
{
  type TransferMode = Value
  val equalize, distribute, drain, fill = Value

  implicit class TransferModeValue(value: Value)
  {
    def toggle: TransferMode = TransferMode((value.id + 1) % TransferMode.values.size)
  }

}

File: src/main/scala/mffs/util/TransferMode.java
package mffs.util;

/**
 * @author Calclavia
 */
public enum TransferMode {
	equalize, distribute, drain, fill;

	public TransferMode toggle() {
		return TransferMode.values()[ordinal() + 1 % TransferMode.values().length];
	}
}
